### PR TITLE
Add ITs for JVM old gen policy

### PR DESCRIPTION
**Pull request by rguo-aws**
…Rest endpoint

*Fixes #:*

*Description of changes:*
Add ITs for JVM old gen policy. This PR fixes a few minor issues in IT framework to allow IT to retrieve persisted action list from rest endpoint. We also add six ITs to cover the happy scenario of old gen policy in JVM decider. 
Those ITs are :
ITs that triggers LevelOne/LevelTwo/LevelThree action builder (dedicated master / multinode)

*Tests:*

*If new tests are added, how long do the new ones take to complete*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java [hunk 1]
```diff
@@ -278,7 +278,7 @@ synchronized int insertRow(String tableName, List<Object> row) throws SQLExcepti
       recordsWithMaxFieldValue = create.select().from(tableName).where(DSL.field(field)
               .eq(create.select(max(field)).from(tableName))).fetch();
     } catch (DataAccessException dex) {
-      LOG.error("Error querying table {}", tableName, dex);
+      //LOG.warn("Error querying table {}", tableName, dex);
```
Comment (yojs) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java:281]: Should we just remove it ?

**Review by yojs — DISMISSED** [review]

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/CacheClearAction.java [hunk 2]
```diff
@@ -120,4 +118,39 @@ public CacheClearAction build() {
       return new CacheClearAction(appContext, coolOffPeriodInMillis, canUpdate);
     }
   }
+
+  public static class Summary {
```
Comment (sruti1312) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/CacheClearAction.java:122]: Can we make this as an util class or rename as CacheClearActionSummary?

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvm/old_gen_policy/dedicated_master/LevelThreeDedicatedMasterITest.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      reason = "Shard request cache metrics is expected to be missing")
+  @AErrorPatternIgnored(
+      pattern = "CacheUtil:getCacheMaxSize()",
+      reason = "Shard request cache metrics is expected to be missing.")
+  @AErrorPatternIgnored(
+      pattern = "HighHeapUsageYoungGenRca:operate()",
+      reason = "YoungGen metrics is expected to be missing.")
+  @AErrorPatternIgnored(
+      pattern = "PersistableSlidingWindow:<init>()",
+      reason = "Persistence base path can be null for integration test.")
```
Comment (sruti1312) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvm/old_gen_policy/dedicated_master/LevelThreeDedicatedMasterITest.java:175]: This is fixed with Sid's PR. We dont need to add them here

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/CacheClearAction.java (hunk 2 above)
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/CacheClearAction.java:122]: Since this Summary is only used as a Json container within each action class(similar to the Builder class within the action class), I would like to leave it as it is. We can make them as separate classes if needed in the future

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvm/old_gen_policy/dedicated_master/LevelThreeDedicatedMasterITest.java (hunk 3 above)
Comment (rguo-aws) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvm/old_gen_policy/dedicated_master/LevelThreeDedicatedMasterITest.java:175]: thanks for catching this. Yes. let me remove those in my next PRs(I am creating a few more ITs to cover corner cases in JVM decider)

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/OldGenRca.java [hunk 4]
```diff
@@ -72,7 +73,7 @@ protected double getMaxOldGenSizeOrDefault(final double defaultValue) {
       }
       double ret =
           SQLParsingUtil
-              .readDataFromSqlResult(heapMaxMetric.getData(), MEM_TYPE.getField(), OLD_GEN.toString(), MetricsDB.MAX);
+              .readDataFromSqlResult(heapMaxMetric.getData(), MEM_TYPE.getField(), HEAP.toString(), MetricsDB.MAX);
```
Comment (sruti1312) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/OldGenRca.java:76]: I see that you have updated OLD_GEN to HEAP. The method says getMaxOldGenSizeOrDefault. Is there a reason why this change was made? Do we want to rename the method?

Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/OldGenRca.java:76]: yes, this was a bug I found in old gen RCA. Heap usage is supposed to be calculated by (old gen usage)/(max total heap size). but for some reason, we calculated it by (old gen usage)/(max old gen size)

**Review by sruti1312 — DISMISSED** [review]

**Comment by sruti1312** [comment]
com.amazon.opendistro.elasticsearch.performanceanalyzer.store.rca.hotheap.HighHeapUsageOldGenRcaTest

  Test testHighHeapOldGenRca FAILED

  java.lang.AssertionError

There are some tests failing

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvm/old_gen_policy/dedicated_master/LevelThreeDedicatedMasterITest.java (hunk 3 above)
Comment (rguo-aws) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvm/old_gen_policy/dedicated_master/LevelThreeDedicatedMasterITest.java]: done, removed this pattern

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/OldGenRca.java (hunk 4 above)
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/OldGenRca.java]: done. added a new function to retrieve HEAP and leave the OLD_GEN unchanged

**Review by sruti1312 — APPROVED** [review]

**Review by yojs — APPROVED** [review]
